docs/src/userguide/language_basics.rst

```diff
@@ -687,19 +687,24 @@ as a contract with the caller. Here is an example:
             @cython.cfunc
             @cython.exceptval(-1)
             def spam() -> cython.int:
-                ...
+                raise ValueError()
 
     .. group-tab:: Cython
 
         .. code-block:: cython
 
             cdef int spam() except -1:
-                ...
+                raise ValueError()
+
 
 With this declaration, whenever an exception occurs inside ``spam``, it will
 immediately return with the value ``-1``.  From the caller's side, whenever
 a call to spam returns ``-1``, the caller will assume that an exception has
-occurred and can now process or propagate it.
+occurred and can now process or propagate it. Calling ``spam()`` is roughly equivalent to::
+
+    ret_val = spam()
+    if ret_val == -1:
+        raise ValueError()
```

Comment:
I'm not sure this is quite right. It's closer to `if ret_val == -1: raise` (i.e. raise whatever error is currently existing).

I don't think this quite translates into Python pseudo-code though. This feels to me like having something a little wrong could cause more confusion than it solves.

I'd actually be happier with a simplified version of the generated C code:

```
ret_val = spam();
if (ret_val == -1) goto error_label;
```

but then this pre-supposes some knowledge of the Python C API, so may not be great.

In short I don't quite know what should go here.

Reply:
> I'm not sure this is quite right. 

Yes these examples are simplified to express the high level logic. My ultimate goal was to do something similar to the python itertools documentation [1]. If there is consensus that this is wrong direction feel free to close this PR or just I can remove the controversial parts. I am also fine with converting it to the simplified C code. The examples are pretty simple so I think that even people not knowing C syntax will be able to extract the main point. (BTW. in my opinion any non-trivial usage of Cython requires at least basic understanding of C)


[1] https://docs.python.org/3/library/itertools.html#itertools.accumulate

Reply:
@da-woods I have migrated examples to C as you proposed.

Reply:
Question: maybe as a small improvement we can use `error_handler` instead of `error_label` to improve readability of example. What do you think?

Reply:
I think C is probably a clearer way of showing it - with Python I was just getting hung up on the idea of an unhandled exception being there without a try/catch. It just didn't write make sense to me.

I think `error_handler` is better, yes.

Reply:
OK everything implemented.